<a href="https://colab.research.google.com/github/mandyzhi/housing-data/blob/main/housing_data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
from google.colab import drive

In [5]:
drive.mount('/content/drive')

Mounted at /content/drive


In [6]:
import pandas

housing_data = pandas.read_csv('/content/drive/MyDrive/housing_price_dataset.csv')

In [ ]:
housing_data.describe()

SquareFeet      Bedrooms     Bathrooms     YearBuilt          Price
count  50000.000000  50000.000000  50000.000000  50000.000000   50000.000000
mean    2006.374680      3.498700      1.995420   1985.404420  224827.325151
std      575.513241      1.116326      0.815851     20.719377   76141.842966
min     1000.000000      2.000000      1.000000   1950.000000  -36588.165397
25%     1513.000000      3.000000      1.000000   1967.000000  169955.860225
50%     2007.000000      3.000000      2.000000   1985.000000  225052.141166
75%     2506.000000      4.000000      3.000000   2003.000000  279373.630052
max     2999.000000      5.000000      3.000000   2021.000000  492195.259972

The average price of the houses is $224,827.33

In [ ]:
number_of_houses = len(housing_data)
print(number_of_houses)

50000


1. The number of houses in this dataset is 50,000 houses.

In [ ]:
two_more_bedroom = len(housing_data[housing_data["Bedrooms"] >= 2])

print(two_more_bedroom)

50000


2. All houses have at least two bedrooms.

In [20]:
at_least_two_bathrooms = housing_data[housing_data["Bathrooms"] >= 2]

print(at_least_two_bathrooms["Bedrooms"].sum() / len(at_least_two_bathrooms))




3.5038953226049028


3. The average number of bedrooms, houses with 2 or more bathrooms have is 3.5 bedrooms.

In [27]:
housing_data.sort_values(by= ["Price"]).tail(10)

SquareFeet  Bedrooms  Bathrooms Neighborhood  YearBuilt          Price
2845         2999         5          2        Urban       1999  461502.014630
25017        2970         4          3        Urban       2005  462557.562958
49924        2830         3          1       Suburb       1983  463977.238482
26973        2771         5          2        Urban       1960  467492.827823
11322        2888         2          2        Urban       1953  468470.378004
21815        2998         3          2        Urban       1995  468493.877841
3784         2901         2          3        Urban       1996  470989.679074
26930        2995         5          2        Urban       1954  476671.733263
46610        2957         5          1       Suburb       2001  482577.163405
7430         2758         5          2        Rural       1967  492195.259972

4. The square feet of the ten most expensive houses, in the order of most expensive to least, are:

  1. 2758 square feet
  2. 2957 square feet
  3. 2995 square feet
  4. 2901 square feet
  5. 2998 square feet
  6. 2888 square feet
  7. 2771 square feet
  8. 2830 square feet
  9. 2970 square feet
  10. 2999 square feet

In [32]:
housing_prices = housing_data.sort_values(by=["Price"])

print(housing_prices.head(1))
print(housing_prices.tail(1))

       SquareFeet  Bedrooms  Bathrooms Neighborhood  YearBuilt         Price
33666        1013         5          2        Urban       1960 -36588.165397
      SquareFeet  Bedrooms  Bathrooms Neighborhood  YearBuilt          Price
7430        2758         5          2        Rural       1967  492195.259972


5. The most expensive house was created in year 1967, and the cheapest house was created in year 1960.

In [33]:
urban_houses = housing_data[housing_data["Neighborhood"] == "Urban"]

urban_houses_average = urban_houses["SquareFeet"].sum() / len(urban_houses)

print(urban_houses_average)

2017.4169126061556


6. The average size of Urban houses is 2017.4 square feet.

In [34]:
rural_houses = housing_data[housing_data["Neighborhood"] == "Rural"]

rural_houses_average = rural_houses["SquareFeet"].sum() / len(rural_houses)

print(rural_houses_average)

2001.621132165987


7. The average size of Rural houses is 2001.6 square feet.

In [39]:
one_bathroom = housing_data[housing_data["Bathrooms"] == 1]
print(len(one_bathroom)/len(housing_data))

0.3351


8. 33% of houses have one bathroom.

In [48]:
two_two = housing_data[(housing_data["Bedrooms"] == 2) & (housing_data["Bathrooms"] == 2)]

print(len(two_two))

4183


9. 4183 houses have two bedrooms and two bathrooms.

In [49]:
two_bathrooms = housing_data[housing_data["Bathrooms"] == 2]

two_bathroom_avg = two_bathrooms["Price"].sum() / len(two_bathrooms)

print(two_bathroom_avg)

225291.3031288047


10. The average price for houses with two bathrooms is 225,291.30.